prov/efa/configure.m4

```diff
@@ -151,40 +148,32 @@ AC_DEFUN([FI_EFA_CONFIGURE],[
 			[$efa_LIBDIR],
 			[],
 			[efadv_support_extended_cq=0])
-	      ])
 
-	# efadv_wc_read_sgid is a static inline function, which is
-	# required to use extended CQ for recovering peer address.
-	# Extended CQ should be disabled if it is not available.
-	AS_IF([test $efadv_support_extended_cq -eq 1],
-		[AC_EGREP_HEADER(
-			[int efadv_wc_read_sgid],
-			[infiniband/efadv.h],
+		AC_CHECK_DECL([efadv_wc_read_sgid],
 			[],
-			[efadv_support_extended_cq=0])
-		])
-
-	AS_IF([test $efadv_support_extended_cq -eq 1],
-		[AC_DEFINE([HAVE_EFADV_CQ_EX], [1], [EFA device support extensible CQ])],
-		[AC_DEFINE([HAVE_EFADV_CQ_EX], [0], [EFA device does not support extensible CQ])]
-		)
+			[efadv_support_extended_cq=0],
+			[[#include <infiniband/efadv.h>]])
+	])
 
-	AM_CONDITIONAL([HAVE_EFADV_CQ_EX], [test $efadv_support_extended_cq -eq 1])
-
-	# IBV_QUERY_QP_DATA_IN_ORDER_ALIGNED_128_BYTES is the flag we need test,
-	# but it is flag. If it is not there, we should disable it.
-	AS_IF([test $efa_support_data_in_order_aligned_128_byte -eq 1],
-		[AC_EGREP_HEADER(
-			[IBV_QUERY_QP_DATA_IN_ORDER_ALIGNED_128_BYTES],
-			[infiniband/verbs.h],
-			[],
-			[efa_support_data_in_order_aligned_128_byte=0])
-		])
+	AC_DEFINE_UNQUOTED([HAVE_RDMA_SIZE],
+		[$have_rdma_size],
+		[Indicates if efadv_device_attr has max_rdma_size])
+	AC_DEFINE_UNQUOTED([HAVE_CAPS_RNR_RETRY],
+		[$have_caps_rnr_retry],
+		[Indicates if EFADV_DEVICE_ATTR_CAPS_RNR_RETRY is defined])
+	AC_DEFINE_UNQUOTED([HAVE_CAPS_RDMA_WRITE],
+		[$have_caps_rdma_write],
+		[Indicates if EFADV_DEVICE_ATTR_CAPS_RDMA_WRITE is defined])
+	AC_DEFINE_UNQUOTED([HAVE_IBV_IS_FORK_INITIALIZED],
+		[$have_ibv_is_fork_initialized],
+		[Indicates if libibverbs has ibv_is_fork_initialized])
+	AC_DEFINE_UNQUOTED([HAVE_EFADV_CQ_EX],
+		[$efadv_support_extended_cq],
+		[Indicates EFA supports extensible CQ])
+	AC_DEFINE_UNQUOTED([HAVE_EFA_DATA_IN_ORDER_ALIGNED_128_BYTES],
+		[$efa_support_data_in_order_aligned_128_byte],
+		[Indicates if EFA supports 128 bytes in-order in writing.])
 
-	AS_IF([test $efa_support_data_in_order_aligned_128_byte -eq 1],
-		[AC_DEFINE([HAVE_EFA_DATA_IN_ORDER_ALIGNED_128_BYTES], [1], [EFA device supports 128 bytes in-order in writing.])],
-		[AC_DEFINE([HAVE_EFA_DATA_IN_ORDER_ALIGNED_128_BYTES], [0], [EFA device does not support 128 bytes in-order in writing.])]
-		)
 
 	efa_CPPFLAGS="$efa_ibverbs_CPPFLAGS $efadv_CPPFLAGS"
```

Comment:
I would put these further down, after all the tests (even though they're unit test related tests, someone else might miss that distinction).

Reply:
I'm not sure which lines you are referring to here.